Climate_police/homework 7/temperature.py

```diff
@@ -0,0 +1,178 @@
+        http = urllib3.PoolManager(
+            cert_reqs='CERT_REQUIRED', # Force certificate check.
+            ca_certs=certifi.where(),  # Path to the Certifi bundle.
+            retries=3
+        )
+
+        try:
+            response = http.request('GET',URL)
+            with open(filename,'wb') as f:
+                f.write(response.data)
+            f.close()
+            response.release_conn()
+
+        except:
+            e = sys.exc_info()[0]
+            print('Error! Data not downloaded.',e)
+
+def get_temperature_data():
+    """
+    Download temperature data, unless already downloaded
+    """
+    download_if_needed('https://www.kaggle.com/berkeleyearth/climate-change-earth-surface-temperature-data/downloads/GlobalLandTemperatures.zip',
+    'GlobalLandTemperatures.zip')
+
+def global_temp_country():
+    """
+    Fetch data (if needed) and extract global temperatures of countries
+    """
+    get_temperature_data()
+    zf = zipfile.ZipFile('GlobalLandTemperatures.zip')
+    file_handle = zf.open('GlobalLandTemperaturesByCountry.csv')
+    return pd.read_csv(file_handle)
+
+def mercator():
+    """
+    Creates Mercator projection of the world Map tos how temperature
+    """
+    data = [ dict(
+        type = 'choropleth',
+        locations = countries,
+        z = mean_temp,
+        text = countries,
+        locationmode = 'country names',
+        autocolorscale = True,
+        reversescale = False,
+        marker = dict(
+            line = dict (
+                color = 'rgb(180,180,180)',
+                width = 0.5
+            ) ),
+        colorbar = dict(
+            autotick = False,
+            title = 'Average land temperature'),
+      ) ]
+
+    layout = dict(
+        title = 'Average land temperature by country',
+        geo = dict(
+            showframe = False,
+            showcoastlines = False,
+            showocean = True,
+            oceancolor = 'rgb(10,200,255)',
+            projection = dict(
+                type = 'Mercator'
+                )
+            )
+        )
+
+    fig = dict( data=data, layout=layout )
+    py.iplot( fig, validate=False, filename='d3-world-map' )
+
+def globe_graph():
+    """
+    Cleaning data & plotting average land temperature
+    """
+    global_temp_country_clean = global_temp_country[~global_temp_country['Country'].isin(
+    ['Denmark', 'Antarctica', 'France', 'Europe', 'Netherlands',
+     'United Kingdom', 'Africa', 'South America'])]
+
+    global_temp_country_clean = global_temp_country_clean.replace(
+       ['Denmark (Europe)', 'France (Europe)', 'Netherlands (Europe)', 'United Kingdom (Europe)'],
+       ['Denmark', 'France', 'Netherlands', 'United Kingdom'])
+
+    #Let's average temperature for each country
+
+    countries = np.unique(global_temp_country_clean['Country'])
+    mean_temp = []
+    for country in countries:
+        mean_temp.append(global_temp_country_clean[global_temp_country_clean['Country'] ==
```

Comment:
I would also consider the maximum value of temperature, since it could be a significant data to visualize!!